database/index.js

```diff
@@ -11,17 +11,36 @@ let morePlaces = mongoose.Schema({
   rating: Number,
   numOfRatings: Number,
   description: String,
-  price: Number
+  price: Number,
+  similarPlaces: Array
 });
 
 let relatedPlaces = mongoose.model('MorePlaces', morePlaces);
 
 
+
 let getAllProperties = (callback) => {
-  relatedPlaces.find( (err, results) => {
+  relatedPlaces.find( {}, (err, results) => {
     if(err) {
       callback(err);
     } else {
+      let length = results.length;
+      for (let i = 0; i < length; i++) {
+        for (let j = 0; j < 12; j++) {
+          // console.log(results, 'results ')
+          // console.log(randomNum)
+          let recursive = () => {
+            let randomNum = Math.floor(Math.random() * Math.floor(results.length))
+            if (randomNum === i) {
+              return recursive();
+            } else {
+            results[i].similarPlaces.push(results[randomNum]._id);
+            }
+          }
+          recursive();
```

Comment:
wow using a recursion here. I am not quite understanding the logic here. curious what its doing